isis/src/base/objs/PvlKeyword/PvlKeyword.cpp

```diff
@@ -1391,8 +1393,24 @@ namespace Isis {
           KEYWORD
 
         Well, no value/units may exist so we're done processing this keyword.
+        Now check if it's a reserved keyword that is valueless
        */
-      return 1; // Valid & Successful
+      bool allowedValueless = false;
+      static const std::vector<QString> reservedKeywordNames = {"OBJECT",
+                                                                "BEGIN_OBJECT",
+                                                                "END_OBJECT",
+                                                                "ENDOBJECT",
+                                                                "GROUP",
+                                                                "END_GROUP",
+                                                                "ENDGROUP",
+                                                                "END"};
+      const QString keywordNameUpper = keywordName.toUpper();
+      for (const auto &reservedKeywordName : reservedKeywordNames) {
+        if (keywordNameUpper == reservedKeywordName) {
+          allowedValueless = true;
+        }
+      }
+      return allowedValueless;
```

Comment:
@acpaquette so sorry I didn't see this earlier...

You should , at least, insert a `break;` after line 1410 otherwise you run through the entire `reservedKeywordNames` list which isn't necessary.

Reply:
@KrisBecker No worries! I added it after you suggested making the reserved keywords const. So you didn't miss it on initial review